crypto/merklearray/proof_test.go

```diff
@@ -46,18 +43,84 @@ func TestProofSerialization(t *testing.T) {
 
 	// check the padded results
 	zeroDigest := make([]byte, crypto.Sha512_256Size)
-	require.Equal(t, data[1+((MaxEncodedTreeDepth-1)*crypto.Sha512_256Size):], zeroDigest)
-
-	var newPath []crypto.GenericDigest
-	for i := 0; i < MaxEncodedTreeDepth+1; i++ {
-		var junkDigest [crypto.Sha512_256Size]byte
-		crypto.RandBytes(junkDigest[:])
-		newPath = append(newPath, junkDigest[:])
+	i := 0
+	for ; i < (MaxEncodedTreeDepth - 2); i++ {
+		require.Equal(t, zeroDigest, data[1+crypto.Sha512_256Size*i:1+crypto.Sha512_256Size*(i+1)])
```

Comment:
`a := require.New(t)` and then `a.Equal( , )` might be cleaner for the assertions.
Define `a` at the function prologue.